pypit/armed.py

```diff
@@ -177,27 +182,58 @@ def ARMED(fitsdict, reuseMaster=False, reloadMaster=True):
                     slf.SetFrame(slf._satmask, satmask, det)
                     slf.SetFrame(slf._tiltpar, outpar, det)
 
-            artrace.slit_profile(slf, slf._mstrace[det - 1], det)
-            msgs.error("check it")
+            # msgs.info("Temporary code")
+            # sciframe = arload.load_frames(fitsdict, [scidx], det, frametype='science', msbias=slf._msbias[det - 1])
+            # sciframe = sciframe[:, :, 0]
+            # arproc.background_subtraction(slf, sciframe, sciframe, 30, det)
+            # msgs.error("Remove this code")
+
             ###############
             # Prepare the pixel flat field frame
             update = slf.MasterFlatField(fitsdict, det)
             if update and reuseMaster: armbase.UpdateMasters(sciexp, sc, det, ftype="flat", chktype="pixelflat")
 
-            msgs.error("UP TO HERE!!!")
+            ###############
+            # Derive the spatial profile and blaze function
+            if slf._slitprof[det-1] is None:
+                if settings.argflag['reduce']['masters']['reuse']:
+                    msslitprof_name = armasters.master_name('slitprof', settings.argflag['reduce']['masters']['setup'])
+                    try:
+                        slit_profiles, head = arload.load_master(msslitprof_name, frametype="slit profile")
+                    except IOError:
+                        pass
+                    else:
+                        slf.SetFrame(slf._slitprof, slit_profiles, det)
+                        settings.argflag['reduce']['masters']['loaded'].append('slitprof'+settings.argflag['reduce']['masters']['setup'])
+                if 'slitprof'+settings.argflag['reduce']['masters']['setup'] not in settings.argflag['reduce']['masters']['loaded']:
+                    # First time slit profile is derived
+                    msgs.info("Calculating slit profile from master trace frame")
+                    slit_profiles, mstracenrm, msblaze, flat_ext1d = arproc.slit_profile(slf, slf._mstrace[det - 1], det)
+                    slf.SetFrame(slf._slitprof, slit_profiles, det)
+                    slf.SetFrame(slf._msblaze, msblaze, det)
+                    # Prepare some QA for the average slit profile along the slit
+                    msgs.info("Preparing QA of each slit profile")
+                    arqa.slit_profile(slf, mstracenrm, slit_profiles, slf._lordloc[det - 1], slf._rordloc[det - 1],
+                                      slf._slitpix[det - 1], desc="Slit profile")
+                    msgs.info("Saving blaze function QA")
+                    arqa.plot_orderfits(slf, msblaze, flat_ext1d, desc="Blaze function", textplt="Order")
```

Comment:
OK, I've added these if statements... and I'm starting to agree with you about the pdfs. It's something we need to discuss more, and perhaps we'll get some feedback as more people start using the code. Instead of a PDF, perhaps we can create a GUI that is specifically designed for inspecting the QA... Anyway, let's add this to an Issue.